src/robusta/core/discovery/discovery.py

```diff
@@ -169,28 +216,37 @@ def discovery_process() -> DiscoveryResults:
         # discover jobs
         active_jobs: List[JobInfo] = []
         try:
-            current_jobs: V1JobList = client.BatchV1Api().list_job_for_all_namespaces()
-            for job in current_jobs.items:
-                job_pods = []
-                job_labels = {}
-                if job.spec.selector:
-                    job_labels = job.spec.selector.match_labels
-                elif job.metadata.labels:
-                    job_name = job.metadata.labels.get("job-name", None)
-                    if job_name:
-                        job_labels = {"job-name": job_name}
-
-                if job_labels:  # add job pods only if we found a valid selector
-                    job_pods = [
-                        pod.metadata.name
-                        for pod in pod_items
-                        if (
-                            (job.metadata.namespace == pod.metadata.namespace)
-                            and (job_labels.items() <= (pod.metadata.labels or {}).items())
-                        )
-                    ]
+            continue_ref: Optional[str] = None
+            for _ in range(DISCOVERY_MAX_BATCHES):
+                current_jobs: V1JobList = client.BatchV1Api().list_job_for_all_namespaces(
+                    limit=DISCOVERY_BATCH_SIZE, _continue=continue_ref
+                )
+                for job in current_jobs.items:
+                    job_pods = []
+                    job_labels = {}
+                    if job.spec.selector:
+                        job_labels = job.spec.selector.match_labels
+                    elif job.metadata.labels:
+                        job_name = job.metadata.labels.get("job-name", None)
+                        if job_name:
+                            job_labels = {"job-name": job_name}
+
+                    if job_labels:  # add job pods only if we found a valid selector
+                        job_pods = [
+                            pod.metadata.name
+                            for pod in pod_items
```

Comment:
also will not be all pods, pod_items is only last batch of pods

Reply:
might need to move job discovery before pod inorder to detect if a pod is related to a job

Reply:
fixed